yt/utilities/lib/line_integral_convolution.pyx

```diff
@@ -104,5 +104,8 @@ def line_integral_convolution_2d(
                         &x, &y, &fx, &fy, w, h)
                 l-=1
                 result[i,j] += kernel[l]*texture[x,y]
-                    
+
+            if vectors[i,j,0]==0 and vectors[i,j,1]==0:
+                result[i,j] = 0.
```

Comment:
I think this early return condition should come right after the double for loops, there's no need to do the double while loop for these pixels if we're just going to overwrite the result. That should make it clearer that this is an early return condition as well.